sympy/solvers/diophantine.py

```diff
@@ -55,12 +55,12 @@ def diophantine(eq, param=symbols("t", integer=True)):
     >>> from sympy.solvers.diophantine import diophantine
     >>> from sympy.abc import x, y, z
     >>> diophantine(x**2 - y**2)
-    set([(-t_0, -t_0), (t_0, -t_0)])
+    set([(t_0, -t_0), (t_0, t_0)])
```

Comment:
after this change the next run gave

```
3941Expected:

3942    set([(t_0, -t_0), (t_0, t_0)])

3943Got:

3944    set([(-t_0, -t_0), (t_0, -t_0)])

3945**********************************************************************

3946File "/home/travis/virtualenv/python3.3.5/lib/python3.3/site-packages/sympy-1.0.1.dev0-py3.3.egg/sympy/solvers/diophantine.py", line 60, in sympy.solvers.diophantine.diophantine

3947Failed example:

3948    diophantine(x*(2*x + 3*y - z))

3949Expected:

3950    set([(0, n1, n2), (3*t - z, -2*t + z, z)])

3951Got:

3952    set([(0, n1, n2), (t_0, -t_1, 2*t_0 - 3*t_1)])

3953**********************************************************************

3954File "/home/travis/virtualenv/python3.3.5/lib/python3.3/site-packages/sympy-1.0.1.dev0-py3.3.egg/sympy/solvers/diophantine.py", line 62, in sympy.solvers.diophantine.diophantine

3955Failed example:

3956    diophantine(x**2 + 3*x*y + 4*x)

3957Expected:

3958    set([(0, n1), (3*t - 4, -t)])

3959Got:

3960    set([(0, n1), (3*t_0 - 4, -t_0)])
```

so there is some non-determinism present. This sign problem can be fixed by making the first element of a tuple canonically signed and then changing the other appropriately. I wonder if `clear_coefficients` could be used to reduce the first element to a symbol to give all the constants on the second (and latter) elements....